lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -776,9 +791,19 @@ SPIRVValue *LLVMToSPIRV::transValueWithoutDecoration(Value *V,
       MemoryAccess[0] |= MemoryAccessNontemporalMask;
     if (MemoryAccess.front() == 0)
       MemoryAccess.clear();
+
+    SPIRVValue *BSV = nullptr;
+    if(Function *F = dyn_cast<Function>(ST->getValueOperand())) {
```

Comment:
I think it worth checking that `SPV_INTEL_function_pointers` extension is supported before inserting `FunctionPointerINTEL` instruction

Reply:
Maybe we should check it in addFunctionPointerINTELInst() then ?